### PR TITLE
Added pagination params to listing and filtering submissions

### PR DESCRIPTION
**Pull request by nialldonnellyfh**

### CODE REVIEW

File: lib/cmd/fh3/appforms/environments/submissions/list.js [hunk 1]
```diff
@@ -1,16 +1,27 @@
...
+    return "/api/v2/mbaas/" + params.environment + "/appforms/submissions?page=" + params.page + "&limit=" + params.limit;
+  },
+  'method': 'get',
+  'postCmd': function(submissionsResult, cb){
+    submissionsResult = submissionsResult || {};
+    var table = common.createTableForSubmissions(submissionsResult);
+    var submissionsArray = submissionsResult.submissions;
+    submissionsArray._table = table;
+    cb(undefined, submissionsArray);
   },
```
Comment (fh-qube) [inline lib/cmd/fh3/appforms/environments/submissions/list.js]: MAJOR Remove this trailing comma. [rule](https://metrics.skunkhenry.com/coding_rules#rule_key=javascript%3ATrailingComma)

**Comment by fh-qube** [comment]
SonarQube analysis reported 1 issue:
* MAJOR 1 major

Watch the comments in this conversation to review them.

**Comment by fh-automatron** [comment]
### Pull Request Quality Guidelines

As we continue to focus on quality, here are a few things to keep in mind:

* The code changes conform to the contributing & code style guides for this component (if they exist)
* A component owner should review your change (we've tagged these above)
* Automated testing has been added or updated for this change (e.g. UART, Unit/Acceptance tests)
* There should be an associated fhcap PR with this change - please provide a link to this
* If updating server component this PR should be deployed to a grid somewhere (via `fhcap-cli`) - please provide a link to this
* Ensure JIRA is up to date before & after merge (e.g. deployment notes, tester notes, release notes, fix version)
* Any SonarQube issues should be addressed where applicable (use your own judgement)

Please ping a component owner (see `fh contrib`) when your builds are complete, changes are deployed and you're ready for a review.

In order to update fhcap with your component change please execute:
`fhcap component update fh-fhc 2.8.0 290`

File: lib/cmd/fh3/appforms/environments/submissions/filter.js [hunk 2]
```diff
@@ -1,17 +1,20 @@
...
+var common = require('../../../../../common');
 
 module.exports = {
   'desc': 'Filter Submissions By Form Or Project Id',
   'examples': [{
     cmd: 'fhc appforms environments submissions filter --environment=<Environment Id> [projectid=<Filter By Project ID>] [formid=<Filter By Form ID>]',
     desc: 'Filter Submissions By Form Or Project Id'
   }],
-  'demand': ['environment'],
+  'demand': ['environment', 'page', 'limit'],
```
Comment (pb82) [inline lib/cmd/fh3/appforms/environments/submissions/filter.js]: The implementation also accepts `page` without `limit` and vice versa. Is it necessarry to demand them here?

Comment (nialldonnellyfh) [inline lib/cmd/fh3/appforms/environments/submissions/filter.js]: I think so. The API will fall back to defaults, but it makes sense for the user to at least specify them for fhc.

File: lib/cmd/fh3/appforms/environments/submissions/filter.js [hunk 3]
```diff
@@ -1,17 +1,20 @@
...
   'describe': {
     'environment': "Environment ID",
     'projectid': "Filter By Project ID",
-    'formid': "Filter By Form ID"
+    'formid': "Filter By Form ID",
+    'page': "Page number to list",
+    'limit': "Number of submissions per page"
   },
   'url': function (params) {
     return "/api/v2/mbaas/" + params.environment + "/appforms/submissions/filter_submissions";
```
Comment (pb82) [inline lib/cmd/fh3/appforms/environments/submissions/filter.js]: where are the page and limit params in the URL?

Comment (nialldonnellyfh) [inline lib/cmd/fh3/appforms/environments/submissions/filter.js]: In the POST request body.

File: lib/common.js [hunk 4]
```diff
@@ -250,6 +250,21 @@ exports.createTableForApps = function (apps) {
   return table;
 };
 
+exports.createTableForSubmissions = function(submissionsResult){
+  var table = new Table({
+    head: ['ID', 'Form ID', 'Form Name', 'Project GUID', 'Project Name', 'Submitted At'],
```
Comment (pb82) [inline lib/common.js]: Just a nit-pick: the heading says 'Submitted At' but you insert a time difference into the table (fromNow). I would expect a timestamp (or maybe rename to column heading to just 'Submitted'?)

File: lib/cmd/fh3/appforms/environments/submissions/filter.js (hunk 3 above)
Comment (pb82) [inline lib/cmd/fh3/appforms/environments/submissions/filter.js]: Ok, that makes sense

**Comment by pb82** [comment]
LGTM, some comments

File: lib/common.js (hunk 4 above)
Comment (nialldonnellyfh) [inline lib/common.js]: Yes, that makes sense. I will change that

**Pull request merged**